alacritty/src/renderer/rects.rs

```diff
@@ -190,3 +195,208 @@ impl RenderLines {
+}
+
+impl RectRenderer {
+    /// Update program when doing live-shader-reload.
+    pub fn set_program(&mut self, program: RectShaderProgram) {
+        self.program = program;
+    }
+
+    pub fn new() -> Result<Self, renderer::Error> {
+        let mut vao: GLuint = 0;
+        let mut vbo: GLuint = 0;
+        let program = RectShaderProgram::new()?;
+
+        unsafe {
+            // Allocate buffers.
+            gl::GenVertexArrays(1, &mut vao);
+            gl::GenBuffers(1, &mut vbo);
+
+            gl::BindVertexArray(vao);
+
+            // VBO binding is not part of VAO itself, but VBO binding is stored in attributes.
+            gl::BindBuffer(gl::ARRAY_BUFFER, vbo);
+
+            let mut attribute_offset = 0;
+
+            // Position.
+            gl::VertexAttribPointer(
+                0,                            // Attribute location.
+                2,                            // Component count: 2. x and y.
+                gl::FLOAT,                    // Attribute type: f32
+                gl::FALSE,                    // Not normalized.
+                size_of::<Vertex>() as i32,   // Stride.
+                attribute_offset as *const _, // Offset of Vertex::x is zero.
+            );
+            gl::EnableVertexAttribArray(0);
+            attribute_offset += size_of::<f32>() * 2;
+
+            // Color.
+            gl::VertexAttribPointer(
+                1,                            // Attribute location.
+                4,                            // Component count: 4. r, g, b, a.
+                gl::UNSIGNED_BYTE,            // Attribute type: u8.
+                gl::TRUE,                     // Normalized: 0..255 is mapped to 0..1.
+                size_of::<Vertex>() as i32,   // Stride.
+                attribute_offset as *const _, // Offset of Vertex::color.
+            );
+            gl::EnableVertexAttribArray(1);
+
+            // Reset buffer bindings.
+            gl::BindVertexArray(0);
+            gl::BindBuffer(gl::ARRAY_BUFFER, 0);
+        }
+
+        Ok(Self { vao, vbo, program, vertices: Vec::new() })
+    }
+
+    pub fn draw(&mut self, size_info: &SizeInfo, rects: Vec<RenderRect>) {
+        unsafe {
+            // Bind VAO to enable vertex attribute slots.
+            gl::BindVertexArray(self.vao);
+
+            // Bind VBO only once for buffer data upload only.
+            gl::BindBuffer(gl::ARRAY_BUFFER, self.vbo);
+
+            gl::UseProgram(self.program.id);
+        }
+
+        let center_x = size_info.width() / 2.;
+        let center_y = size_info.height() / 2.;
+
+        // Build rect vertices vector.
+        self.vertices.clear();
+        for rect in &rects {
+            self.add_rect(center_x, center_y, rect);
+        }
+
+        unsafe {
+            // Upload accumulated vertices.
+            gl::BufferData(
+                gl::ARRAY_BUFFER,
+                (self.vertices.len() * std::mem::size_of::<Vertex>()) as isize,
```

Comment:
In renderer/mod.rs we import `std::mem::size_of` completely, usually I'd discourage that in Alacritty for consistency. But just an import of `std::mem` at the top should help shorten things a little bit.